### DISCUSSION TITLE
Buildpacks not getting detected through uri

### DISCUSSION THREAD

**Discussion by AshwinSridharan0410** [paketo-buildpacks/feedback#257] [Builder Team]
[[buildpacks]]
id = "paketo-buildpacks/web-servers"
version = "0.17.2"
uri = "../web-server-buildpack"
When I pass my buildpacks in the builder like this, it is not getting detected (not participating).

[[buildpacks]]
  uri = "docker://xyz/paketo-buildpacks/web-servers:0.17.2"
  version = "0.17.2"

But when I pass my buildpacks like this , in the builder , it is participating.

This case would be useful , if I have buildpacks of different versions needing to be supported in the same builder image.

Thank you

**Comment by dmikusa** [comment]
I'm not sure why you'd see a difference here based on the info presented. I would suggest pointing to images, but that's not a requirement. I suggest that because images are a lot more precise than a folder. The folder contents can change, whereas the image cannot change (at least not without causing the hash to change).

**Comment by AshwinSridharan0410** [comment]
@dmikusa , yeah the image is participating correctly, but the uri (folder) is not participating.
My reason to use both uri(folder) and image is , like in the image I want some specific versions of node like 14,16,18 and in the uri, I want node version 20.
As the paketo buildpacks is'nt having a release tag that supports all the 4 node versions (14,16,18,20)in a single buildpack.

**Reply by dmikusa** [reply]
> My reason to use both uri(folder) and image is , like in the image I want some specific versions of node like 14,16,18 and in the uri, I want node version 20.
> As the paketo buildpacks is'nt having a release tag that supports all the 4 node versions (14,16,18,20)in a single buildpack.

That's fine and you can do that with images. That is what I suggested on your other thread, https://github.com/orgs/paketo-buildpacks/discussions/255#discussioncomment-8483989.

If you really, really want to use a directory then you need to post more information and include the output of your verbose build (`pack build -v -e BP_DEBUG=true ...`). There are a lot of reasons why detect can fail, but having the output should show more clues why.

**Comment by AshwinSridharan0410** [comment]
Okay @dmikusa . Will share the logs for it
